### DISCUSSION TITLE
工作目录.halo2下为空白

### DISCUSSION THREAD

**Discussion by Lorikein12138** [halo-dev/halo#7112] [Q&A]
### 前置条件

- [x] 已经在 [issues](https://github.com/halo-dev/halo/issues) 列表中搜索了相关问题。
- [x] 这是 Halo 项目本身存在的问题，如果是非项目本身的问题（如：安装部署问题），建议在 [Discussions](https://github.com/halo-dev/halo/discussions) 提交。
- [x] 已经尝试过停用所有的插件，排除是插件导致的问题。
- [x] 如果是插件和主题的问题，请在对应的插件和主题仓库提交。

### 系统信息

- 外部访问地址: https://blog.lorikein.cn/
- 启动时间: 2024-11-30 17:34
- 版本: 2.20.10
- 构建时间: 2024-11-26 12:32
- Git Commit: 033a77e
- Java: IBM Semeru Runtime Open Edition / 21.0.5+11-LTS
- 数据库: Source distribution / 5.7.43-log
- 操作系统: Linux / 6.1.0-10-amd64
- 已激活主题: [Sakura 2.4.1](https://github.com/LIlGG/halo-theme-sakura.git)
- 已启动插件:
  - [live2d 看板娘 1.2.1](https://github.com/LIlGG/plugin-live2d)
  - [图库管理 1.4.0](https://github.com/halo-sigs/plugin-photos)
  - [链接管理 1.5.0](https://github.com/halo-sigs/plugin-links)
  - [瞬间 1.8.0](https://github.com/halo-sigs/plugin-moments)
  - [Sitemap 1.1.2](https://github.com/halo-dev/plugin-sitemap)
  - [RSS 1.3.0](https://github.com/halo-dev/plugin-feed)
  - [搜索组件 1.6.2](https://github.com/halo-dev/plugin-search-widget)
  - [评论组件 2.5.1](https://github.com/halo-dev/plugin-comment-widget)
  - [应用市场 1.9.0](https://www.halo.run/store/apps/app-VYJbF)

### 使用的哪种方式运行？

Docker Compose

### 发生了什么？

使用宝塔的Docker Compose功能部署的halo
运行后系统的root文件夹无.halo2目录
容器内的root文件夹存在.halo2目录但目录内是空白的
无法修改如application.yaml等配置文件

### 复现步骤

打开Docker Compose
使用[halo.run](https://docs.halo.run/getting-started/install/docker-compose?current=external-db)中的部署代码
部署后打开目录发现上方问题

### 相关日志输出

```shell

```

### 附加信息

![Image](https://github.com/user-attachments/assets/c34995f7-e6e4-435d-a721-b2fb80b59f31)
![Image](https://github.com/user-attachments/assets/2021e069-ee4d-45d1-bd44-b5ba717dc6e3)

**Comment by ruibaby** [comment]
> 运行后系统的root文件夹无.halo2目录

官方文档是有映射工作目录到当前目录的，并不在系统的 root 目录。

> 容器内的root文件夹存在.halo2目录但目录内是空白的

没有复现，建议提供完整的 Docker Compose 配置。

/triage needs-information

**Comment by Lorikein12138** [comment]
> > 运行后系统的root文件夹无.halo2目录
> 
> 官方文档是有映射工作目录到当前目录的，并不在系统的 root 目录。
> 
> > 容器内的root文件夹存在.halo2目录但目录内是空白的
> 
> 没有复现，建议提供完整的 Docker Compose 配置。
> 
> /triage needs-information


services:
  halo:
    image: registry.fit2cloud.com/halo/halo:2.20
    restart: on-failure:3
    network_mode: "host"
    volumes:
      - ./halo2:/root/.halo2
    environment:
      # JVM 参数，默认为 -Xmx256m -Xms256m，可以根据实际情况做调整，置空表示不添加 JVM 参数
      - JVM_OPTS=-Xmx256m -Xms256m
    command:
      # 修改为自己已有的 MySQL 配置
      - --spring.r2dbc.url=r2dbc:pool:mysql://localhost:3306/XXXX
      - --spring.r2dbc.username=XXXX
      - --spring.r2dbc.password=XXXX
      - --spring.sql.init.platform=mysql
      # 外部访问地址，请根据实际需要修改
      - --halo.external-url=XXXX
      # 端口号 默认8090
      - --server.port=8090

**Comment by ruibaby** [comment]
未复现此问题，这里的 volume 配置是将工作目录映射到 compose 配置文件的同目录，我不清楚你使用的宝塔是放在哪里的。

![Image](https://github.com/user-attachments/assets/06cd22fe-c16e-4102-b1f7-f0123cc34f50)

容器内也是有这个工作目录的：

![Image](https://github.com/user-attachments/assets/61065be8-b672-43a0-a640-d854da8ad744)

你报告的这个问题看起来并非 Halo 的问题，我将转移到 Discussion。
